bindings/netstandard/ElectionGuard/ElectionGuard.ElectionSetup/ElectionKeyPair.cs

```diff
@@ -72,34 +20,93 @@ public ElectionKeyPair(string ownerId, ulong sequenceOrder, ElGamalKeyPair keyPa
     /// <summary>
     /// The pair of public and private election keys for the guardian
     /// </summary>
-    public EncryptionKeyPair KeyPair { get; set; }
+    public ElGamalKeyPair KeyPair { get; set; }
 
     /// <summary>
     /// The secret polynomial for the guardian
     /// </summary>
     public ElectionPolynomial Polynomial { get; set; }
 
-    public ElectionPublicKey Share()
+    /// <summary>
+    /// Construct an Election Key Pair using a random secret key.
+    /// The polynomial is generated using the random secret key.
+    /// This override is used when the guardian generates the keypair internally.
+    /// </summary>
+    public ElectionKeyPair(
```

Comment:
here we let the constructors do most of the heavy lifting rather than using the `ClassDefinition.StaticMethodInit(withParams)` as is used in python and in c++. This seems more idiomatic for C#.

Reply:
oh, you'll also notice that there are no longer any overrides to pass in test data. Instead we assume that the caller of the constructor can infer the intention from the parameters and documentation. I think this will also make it more clear when rehydrating objects, or any other use cases we may have. The test overrides are moved into the extension class